### PR TITLE
Upgrade OpenVino API to v2.0

### PR DESCRIPTION
**Pull request by kartikeyporwal**
PR for #38:
- Adapts the OpenVINO model inference in `OpenVinoInferenceLearner` from `OpenVINO Inference Engine API v1.0` to `OpenVINO Runtime 2.0`

### CODE REVIEW

File: nebullvm/inference_learners/openvino.py [hunk 1]
```diff
@@ -60,15 +62,19 @@ class OpenVinoInferenceLearner(BaseInferenceLearner, ABC):
 
     def __init__(
         self,
-        exec_network,
+        model: openvino.pyopenvino.Model,
+        compiled_model: openvino.runtime.ie_api.CompiledModel,
+        infer_request: openvino.runtime.ie_api.InferRequest,
```
Comment (jiwaszki) [inline nebullvm/inference_learners/openvino.py:67]: This is an internal way of importing objects without additional enhancements. I would suggest to write it like this:
```suggestion
        model: openvino.runtime.Model,
        compiled_model: openvino.runtime.CompiledModel,
        infer_request: openvino.runtime.InferRequest,
```
This is an actual public API with documentation provided online.

**Review by jiwaszki — COMMENTED** [review]
I am very happy to see new Python API in action! 🥳

I left some minor comments that may improve OpenVINO's part of the project in the future. Please let me know what do you think about them!

File: nebullvm/inference_learners/openvino.py (hunk 1 above)
Comment (kartikeyporwal) [reply nebullvm/inference_learners/openvino.py]: Thanks @jiwaszki for the suggestion. I'd actually ended up using the type inference based on the output of `type` on the respective objects.

File: nebullvm/inference_learners/openvino.py [hunk 2]
```diff
@@ -202,14 +220,20 @@ def _predict_array(
...
-        results = self.exec_network.infer(
+
+        results = self.infer_request.infer(
             inputs={
                 input_key: input_array
                 for input_key, input_array in zip(
                     self.input_keys, input_arrays
                 )
             }
         )
```
[the remark this answers is not in the document]
Comment (kartikeyporwal) [reply nebullvm/inference_learners/openvino.py:231]: Thanks for making us acquainted with the new features of the OV V2.0.

>That means arrays are in the same order as inputs of network, additionally it results in a scenario where all input keys are matched with respective input array.

Yes

I'd read [this guide](https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html#differences-between-inference-engine-and-openvino-runtime-2022-1) where it states that `OpenVINO API (API 2.0)` preserves the input sequence along with other features. I tried to adhere to what was already implemented in this project for `Inference Engine API`. It was more explicit and works with multiple inputs and single input as well.

Please feel free to comment if you suggest the other way. :)

Thanks

[the remark this answers is not in the document]
Comment (jiwaszki) [reply nebullvm/inference_learners/openvino.py:231]: Sure thing! I was just giving general info that may be helpful. However as you stated - it is more explicit way that will be easier for users/contributors to understand in future.

File: nebullvm/inference_learners/openvino.py [hunk 3]
```diff
@@ -126,17 +132,23 @@ def from_model_name(
         """
         if len(kwargs) > 0:
             warnings.warn(f"Found extra parameters: {kwargs}")
-        inference_engine = IECore()
-        network = inference_engine.read_network(
-            model=model_name, weights=model_weights
+
+        core = Core()
+        model = core.read_model(model=model_name, weights=model_weights)
+        compiled_model = core.compile_model(model=model, device_name="CPU")
```
[the remark this answers is not in the document]
Comment (morgoth95) [reply nebullvm/inference_learners/openvino.py:138]: Thanks @jiwaszki for the hint! Actually, we can think of supporting both Intel's GPU and multidevice solutions. We would just have to verify that AUTO mode does not give problems with the quantization process (if so, we would simply have to select the most suitable target depending on the situation).

I would open a new issue to test the AUTO behaviour within the library and then add it in another PR!

[the remark this answers is not in the document]
Comment (kartikeyporwal) [reply nebullvm/inference_learners/openvino.py:138]: Hi @morgoth95,

Does it make to sense to merge this PR till we figure out the `AUTO` mode in the `device_name`?
Also, please feel free to comment what you think about this PR?

Thanks

[the remark this answers is not in the document]
Comment (morgoth95) [reply nebullvm/inference_learners/openvino.py:138]: Hello Kartikey! Did you already reformat the code following the [contribution guidelines](https://github.com/nebuly-ai/nebullvm/blob/main/CONTRIBUTING.md#coding-style)?

**Comment by morgoth95** [comment]
Closes #38

**Pull request merged**